contracts/src/v0.8/EIP_712_1014_4337/consumers/SCA.sol

```diff
@@ -0,0 +1,71 @@
+import "../interfaces/IAccount.sol";
+
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+// Smart Contract Account, a contract deployed for a single user and that allows
+// them to invoke meta-transactions.
+contract SCA is IAccount {
+    uint256 s_nonce;
+    address public immutable s_owner;
+
+    // TODO: move this logic to the Create2Factory, ditch the hardcoding.
+    address constant ENTRY_POINT = 0x0576a174D229E3cFA37253523E645A78A0C91B57;
+
+    // Assign the owner of this contract upon deployment.
+    constructor(address owner) {
+        s_owner = owner;
+    }
+
+    /// @dev Sanity check function.
+    /// @dev TODO: remove.
+    function isSCA() external pure returns (bool) {
+        return true;
+    }
+
+    /// @dev Validates the user operation via a signature check.
+    /// @dev TODO: Add a domain separator to the signature.
+    function validateUserOp(
+        UserOperation calldata userOp,
+        bytes32 userOpHash,
+        uint256 missingAccountFunds
+    ) external returns (uint256 validationData) {
+        s_nonce++;
+        bytes32 h = keccak256(
+            abi.encode(
+                userOp.sender,
+                userOp.callData,
+                s_owner,
+                s_nonce,
+                block.chainid
+            )
+        );
```

Comment:
Relatively minor but this is "non-standard" - `s_nonce` is not incremented and then used but rather used and then incremented (i.e you start from 0, rather than 1).